src/glue/SessionHL.cpp

```diff
@@ -167,6 +172,33 @@ bool SessionHL::Authenticate(const std::string &username, const std::string &pas
   return res;
 }
 
+bool SessionHL::SSOAuthenticate(const std::string &scheme, const std::string &identity_provider_response) {
```

Comment:
I would like to combine the Authenticate and SSOAuthenticate into a single function.
Don't really see why they should be separate.

Reply:
The two methods have now become significantly different